Exception during gitflow-incremental-builder execution occurred. Git branch of name 'HEAD' not found.

I got the above error message when starting gib on a new project via `mvn clean install`.
The repository is fresh (`git init` ) uncommited and has no remote configured yet.
My expectation ist a still buildable project.


Context:
```
git symbolic-ref HEAD
refs/heads/master
```

Debug log:
```
[DEBUG] com.vackosar.gitflowincrementalbuilder:gitflow-incremental-builder:jar:3.11.2
[DEBUG]    org.eclipse.jgit:org.eclipse.jgit:jar:5.8.1.202007141445-r:compile
[DEBUG]       com.googlecode.javaewah:JavaEWAH:jar:1.1.7:compile
[DEBUG]    org.eclipse.jgit:org.eclipse.jgit.ssh.jsch:jar:5.8.1.202007141445-r:compile
[DEBUG]       com.jcraft:jsch:jar:0.1.55:compile
[DEBUG]       com.jcraft:jzlib:jar:1.1.1:compile
[DEBUG]    com.jcraft:jsch.agentproxy.connector-factory:jar:0.0.9:compile
[DEBUG]       com.jcraft:jsch.agentproxy.core:jar:0.0.9:compile
[DEBUG]       com.jcraft:jsch.agentproxy.usocket-jna:jar:0.0.9:compile
[DEBUG]       com.jcraft:jsch.agentproxy.usocket-nc:jar:0.0.9:compile
[DEBUG]    com.jcraft:jsch.agentproxy.jsch:jar:0.0.9:compile
[DEBUG]    com.jcraft:jsch.agentproxy.pageant:jar:0.0.9:compile
[DEBUG]       net.java.dev.jna:jna:jar:4.1.0:compile
[DEBUG]       net.java.dev.jna:jna-platform:jar:4.1.0:compile
[DEBUG]    com.jcraft:jsch.agentproxy.sshagent:jar:0.0.9:compile
[DEBUG]    org.codehaus.plexus:plexus-utils:jar:1.1:runtime
[DEBUG] Created new class realm extension>com.vackosar.gitflowincrementalbuilder:gitflow-incremental-builder:3.11.2
[DEBUG] Importing foreign packages into class realm extension>com.vackosar.gitflowincrementalbuilder:gitflow-incremental-builder:3.11.2
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm extension>com.vackosar.gitflowincrementalbuilder:gitflow-incremental-builder:3.11.2
[DEBUG]   Included: com.vackosar.gitflowincrementalbuilder:gitflow-incremental-builder:jar:3.11.2
[DEBUG]   Included: org.eclipse.jgit:org.eclipse.jgit:jar:5.8.1.202007141445-r
[DEBUG]   Included: com.googlecode.javaewah:JavaEWAH:jar:1.1.7
[DEBUG]   Included: org.eclipse.jgit:org.eclipse.jgit.ssh.jsch:jar:5.8.1.202007141445-r
[DEBUG]   Included: com.jcraft:jsch:jar:0.1.55
[DEBUG]   Included: com.jcraft:jzlib:jar:1.1.1
[DEBUG]   Included: com.jcraft:jsch.agentproxy.connector-factory:jar:0.0.9
[DEBUG]   Included: com.jcraft:jsch.agentproxy.core:jar:0.0.9
[DEBUG]   Included: com.jcraft:jsch.agentproxy.usocket-jna:jar:0.0.9
[DEBUG]   Included: com.jcraft:jsch.agentproxy.usocket-nc:jar:0.0.9
[DEBUG]   Included: com.jcraft:jsch.agentproxy.jsch:jar:0.0.9
[DEBUG]   Included: com.jcraft:jsch.agentproxy.pageant:jar:0.0.9
[DEBUG]   Included: net.java.dev.jna:jna:jar:4.1.0
[DEBUG]   Included: net.java.dev.jna:jna-platform:jar:4.1.0
[DEBUG]   Included: com.jcraft:jsch.agentproxy.sshagent:jar:0.0.9
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:1.1
[DEBUG] Extension realms for project org.example:gib-test:pom:1.0-SNAPSHOT: [ClassRealm[extension>com.vackosar.gitflowincrementalbuilder:gitflow-incremental-builder:3.11.2, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@30946e09]]
[DEBUG] Created new class realm project>org.example:gib-test:1.0-SNAPSHOT
[DEBUG] Populating class realm project>org.example:gib-test:1.0-SNAPSHOT
[DEBUG] Looking up lifecycle mappings for packaging pom from ClassRealm[project>org.example:gib-test:1.0-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]
[DEBUG] Extension realms for project org.example:child1:jar:1.0-SNAPSHOT: [ClassRealm[extension>com.vackosar.gitflowincrementalbuilder:gitflow-incremental-builder:3.11.2, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@30946e09]]
[DEBUG] Looking up lifecycle mappings for packaging jar from ClassRealm[project>org.example:gib-test:1.0-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]
[DEBUG] From default: help=false
[DEBUG] From default: disable=false
[DEBUG] From default: enabled=true
[DEBUG] From default: disableIfBranchRegex=
[DEBUG] From default: disableBranchComparison=false
[DEBUG] From default: referenceBranch=refs/remotes/origin/develop
[DEBUG] From default: fetchReferenceBranch=false
[DEBUG] From default: baseBranch=HEAD
[DEBUG] From default: fetchBaseBranch=false
[DEBUG] From default: useJschAgentProxy=true
[DEBUG] From default: compareToMergeBase=true
[DEBUG] From default: uncommited=true
[DEBUG] From default: untracked=true
[DEBUG] From default: excludePathRegex=
[DEBUG] From default: includePathRegex=
[DEBUG] From default: buildAll=false
[DEBUG] From default: buildAllIfNoChanges=false
[DEBUG] From default: buildDownstream=always
[DEBUG] From default: buildUpstream=derived
[DEBUG] From default: skipTestsForUpstreamModules=false
[DEBUG] From default: argsForUpstreamModules=
[DEBUG] From default: forceBuildModules=
[DEBUG] From default: excludeDownstreamModulesPackagedAs=
[DEBUG] From default: failOnMissingGitDir=true
[DEBUG] From default: failOnError=true
[DEBUG] From default: logImpactedTo=
[INFO] gitflow-incremental-builder 3.11.2 starting...
[DEBUG] file=D:\develop\gib-test\.git\config, create new FileSnapshot: lastRead=2020-07-27 00:07:37.467742100, lastModified=2020-07-26 23:25:55.603789000, size=130, fileKey=java.lang.Object@70211e49
[DEBUG] file=null, size changed from -1 to 244 bytes
[DEBUG] loading config FileBasedConfig[C:\Users\Thomas\.config\jgit\config]
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, create new FileSnapshot: lastRead=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, size=244, fileKey=java.lang.Object@70211e49
[DEBUG] readpipe [git, --version],C:\Program Files\Git\cmd
[DEBUG] readpipe may return 'git version 2.26.2.windows.1'
[DEBUG] remaining output:

[DEBUG] readpipe [git, config, --system, --edit],C:\Program Files\Git\cmd
[DEBUG] readpipe may return 'C:/Program Files/Git/etc/gitconfig'
[DEBUG] remaining output:

[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=null, size changed from -1 to 291 bytes
[DEBUG] loading config FileBasedConfig[C:\Program Files\Git\etc\gitconfig]
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, create new FileSnapshot: lastRead=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, size=291, fileKey=java.lang.Object@70211e49
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, delta=5705036254949000 ns, racy<=2500000000 ns
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, is unmodified
[DEBUG] file=null, size changed from -1 to 59 bytes
[DEBUG] loading config FileBasedConfig[C:\Users\Thomas\.gitconfig]
[DEBUG] file=C:\Users\Thomas\.gitconfig, create new FileSnapshot: lastRead=2020-07-27 00:07:37.596770600, lastModified=2020-05-09 16:54:05.843877000, size=59, fileKey=java.lang.Object@70211e49
[DEBUG] file=D:\develop\gib-test\.git\config, create new FileSnapshot: lastRead=2020-07-27 00:07:37.599771200, lastModified=2020-07-26 23:25:55.603789000, size=130, fileKey=java.lang.Object@70211e49
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, delta=5705036254949000 ns, racy<=2500000000 ns
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.596770600, lastModified=2020-05-09 16:54:05.843877000, delta=6765211752893600 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.gitconfig, is unmodified
[DEBUG] file=D:\develop\gib-test\.git\config, isRacyClean=false, read=2020-07-27 00:07:37.599771200, lastModified=2020-07-26 23:25:55.603789000, delta=2501995982200 ns, racy<=2500000000 ns
[DEBUG] file=D:\develop\gib-test\.git\config, is unmodified
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, delta=5705036254949000 ns, racy<=2500000000 ns
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.596770600, lastModified=2020-05-09 16:54:05.843877000, delta=6765211752893600 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.gitconfig, is unmodified
[DEBUG] file=D:\develop\gib-test\.git\index, create new FileSnapshot: lastRead=2020-07-27 00:07:37.631778700, lastModified=2020-07-26 23:42:09.835439000, size=208, fileKey=java.lang.Object@70211e49
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, delta=5705036254949000 ns, racy<=2500000000 ns
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.596770600, lastModified=2020-05-09 16:54:05.843877000, delta=6765211752893600 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, delta=5705036254949000 ns, racy<=2500000000 ns
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.596770600, lastModified=2020-05-09
16:54:05.843877000, delta=6765211752893600 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, delta=5705036254949000 ns, racy<=2500000000 ns
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.596770600, lastModified=2020-05-09 16:54:05.843877000, delta=6765211752893600 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.gitconfig, is unmodified
[DEBUG] file=D:\develop\gib-test\.git\config, isRacyClean=false, read=2020-07-27 00:07:37.599771200, lastModified=2020-07-26 23:25:55.603789000, delta=2501995982200 ns, racy<=2500000000 ns
[DEBUG] file=D:\develop\gib-test\.git\config, is unmodified
[DEBUG] file=D:\develop\gib-test\.git\HEAD, create new FileSnapshot: lastRead=2020-07-27 00:07:37.720798900, lastModified=2020-07-26 23:25:55.599932000, size=23, fileKey=java.lang.Object@70211e49
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, isRacyClean=false, read=2020-07-27 00:07:37.480744200, lastModified=2020-07-26 23:57:38.796621000, delta=598684123200 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.config\jgit\config, is unmodified
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.581768000, lastModified=2020-05-21 23:23:41.326819000, delta=5705036254949000 ns, racy<=2500000000 ns
[DEBUG] file=C:\Program Files\Git\etc\gitconfig, is unmodified
[DEBUG] file=C:\Users\Thomas\.gitconfig, isRacyClean=false, read=2020-07-27 00:07:37.596770600, lastModified=2020-05-09 16:54:05.843877000, delta=6765211752893600 ns, racy<=2500000000 ns
[DEBUG] file=C:\Users\Thomas\.gitconfig, is unmodified
[DEBUG] file=D:\develop\gib-test\.git\config, isRacyClean=false, read=2020-07-27 00:07:37.599771200, lastModified=2020-07-26 23:25:55.603789000, delta=2501995982200 ns, racy<=2500000000 ns
[DEBUG] file=D:\develop\gib-test\.git\config, is unmodified
[DEBUG] file=D:\develop\gib-test\.git\HEAD, isRacyClean=false, read=2020-07-27 00:07:37.720798900, lastModified=2020-07-26 23:25:55.599932000, delta=2502120866900 ns, racy<=2507500 ns
[DEBUG] file=D:\develop\gib-test\.git\HEAD, is unmodified
[ERROR] Exception during gitflow-incremental-builder execution occurred. Git branch of name 'HEAD' not found. -> [Help 1]
org.apache.maven.MavenExecutionException: Exception during gitflow-incremental-builder execution occurred.
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.perform (MavenLifecycleParticipant.java:85)
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.afterProjectsRead (MavenLifecycleParticipant.java:63)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.IllegalArgumentException: Git branch of name 'HEAD' not found.
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles$Worker.getBranchCommit (DifferentFiles.java:183)
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles$Worker.getBranchDiff (DifferentFiles.java:99)
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles$Worker.access$200 (DifferentFiles.java:84)
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles.get (DifferentFiles.java:61)
    at com.vackosar.gitflowincrementalbuild.control.ChangedProjects.get (ChangedProjects.java:32)
    at com.vackosar.gitflowincrementalbuild.boundary.UnchangedProjectsRemover.act (UnchangedProjectsRemover.java:63)
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.perform (MavenLifecycleParticipant.java:78)
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.afterProjectsRead (MavenLifecycleParticipant.java:63)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
```

root pom:
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>org.example</groupId>
    <artifactId>gib-test</artifactId>
    <packaging>pom</packaging>
    <version>1.0-SNAPSHOT</version>
    <modules>
        <module>child1</module>
    </modules>

    <properties>
        <maven.compiler.source>11</maven.compiler.source>
        <maven.compiler.target>11</maven.compiler.target>
    </properties>

    <build>
    <plugins>
        <plugin>
            <groupId>com.vackosar.gitflowincrementalbuilder</groupId>
            <artifactId>gitflow-incremental-builder</artifactId>
            <version>3.11.2</version>
            <extensions>true</extensions>
        </plugin>
    </plugins>

    </build>
</project>
```

--- a/src/main/java/com/vackosar/gitflowincrementalbuild/control/DifferentFiles.java
+++ b/src/main/java/com/vackosar/gitflowincrementalbuild/control/DifferentFiles.java
@@ -4,11 +4,14 @@ import com.vackosar.gitflowincrementalbuild.boundary.Configuration;
 import com.vackosar.gitflowincrementalbuild.control.jgit.AgentProxyAwareJschConfigSessionFactory;
 import com.vackosar.gitflowincrementalbuild.control.jgit.GitProvider;
 import com.vackosar.gitflowincrementalbuild.control.jgit.HttpDelegatingCredentialsProvider;
+import com.vackosar.gitflowincrementalbuild.entity.SkipExecutionException;
+
 import org.eclipse.jgit.api.FetchCommand;
 import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.api.Status;
 import org.eclipse.jgit.api.errors.GitAPIException;
 import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.Repository;
 import org.eclipse.jgit.revwalk.RevCommit;
 import org.eclipse.jgit.revwalk.RevWalk;
 import org.eclipse.jgit.revwalk.filter.RevFilter;
@@ -177,12 +180,21 @@ public class DifferentFiles {
         }
 
         private RevCommit getBranchCommit(String branchName) throws IOException {
-            ObjectId objectId = git.getRepository().resolve(branchName);
+            Repository repository = git.getRepository();
+            ObjectId objectId = repository.resolve(branchName);
 
             if (objectId == null) {
+                if (repository.simplify(branchName) != null) {
+                    throw new SkipExecutionException("Could not get Git ObjectId for branch of name '" + branchName
+                            + "'. Is this repository empty (no commits yet)?");
+                }
+                if (branchName.startsWith(REFS_REMOTES) && repository.getRemoteNames().isEmpty()) {
+                    throw new SkipExecutionException("Could not get Git ObjectId for branch of name '" + branchName
+                            + "'. No remotes found at all, push still pending?");
+                }
                 throw new IllegalArgumentException("Git branch of name '" + branchName + "' not found.");
             }
-            final RevWalk walk = new RevWalk(git.getRepository());
+            final RevWalk walk = new RevWalk(repository);
             RevCommit commit = walk.parseCommit(objectId);
             walk.close();
             logger.info("Reference commit of branch " + branchName + " is commit of id: " + commit.getId());

--- a/src/test/java/com/vackosar/gitflowincrementalbuild/control/DifferentFilesTest.java
+++ b/src/test/java/com/vackosar/gitflowincrementalbuild/control/DifferentFilesTest.java
@@ -222,8 +222,19 @@ public class DifferentFilesTest extends BaseDifferentFilesTest {
     }
 
     @Test
+    public void invalidBaseBranch() throws Exception {
+        projectProperties.setProperty(Property.referenceBranch.prefixedName(), "FOO");
+
+        Throwable thrown = catchThrowable(() -> invokeUnderTest());
+
+        assertThat(thrown).isInstanceOf(IllegalArgumentException.class);
+        assertThat(thrown).hasMessage("Git branch of name 'FOO' not found.");
+    }
+
+    @Test
     public void emptyLocalRepo() throws Exception {
-        try (InputStream basicPomData = DifferentFilesTest.class.getResourceAsStream("/DifferentFilesTest/pom.xml"); 
+        projectProperties.clear();  // special tests that does not follow the pattern of the test base classes
+        try (InputStream basicPomData = DifferentFilesTest.class.getResourceAsStream("/DifferentFilesTest/pom.xml");
                 EmptyLocalRepoMock emptyLocalRepoMock = new EmptyLocalRepoMock(getRepoBaseFolder())) {
             File projectFolder = emptyLocalRepoMock.getBaseCanonicalBaseFolder();
             Files.copy(basicPomData, new File(projectFolder, "pom.xml").toPath(), StandardCopyOption.REPLACE_EXISTING);
@@ -231,14 +242,15 @@ public class DifferentFilesTest extends BaseDifferentFilesTest {
 
             Throwable thrown = catchThrowable(() -> invokeUnderTest(mavenSessionMock));
 
-            assertThat(thrown).isInstanceOf(IllegalArgumentException.class);
-            assertThat(thrown).hasMessage("Git branch of name 'HEAD' not found.");
+            assertThat(thrown).isInstanceOf(SkipExecutionException.class);
+            assertThat(thrown).hasMessageContaining("'HEAD'");
         }
     }
 
     @Test
     public void localRepoButNoRemoteRepo() throws Exception {
-        try (InputStream basicPomData = DifferentFilesTest.class.getResourceAsStream("/DifferentFilesTest/pom.xml"); 
+        projectProperties.clear();  // special tests that does not follow the pattern of the test base classes
+        try (InputStream basicPomData = DifferentFilesTest.class.getResourceAsStream("/DifferentFilesTest/pom.xml");
                 EmptyLocalRepoMock emptyLocalRepoMock = new EmptyLocalRepoMock(getRepoBaseFolder())) {
             File projectFolder = emptyLocalRepoMock.getBaseCanonicalBaseFolder();
             Files.copy(basicPomData, new File(projectFolder, "pom.xml").toPath(), StandardCopyOption.REPLACE_EXISTING);
@@ -247,9 +259,9 @@ public class DifferentFilesTest extends BaseDifferentFilesTest {
             MavenSession mavenSessionMock = MavenSessionMock.get(projectFolder.toPath(), projectProperties);
 
             Throwable thrown = catchThrowable(() -> invokeUnderTest(mavenSessionMock));
-            
-            assertThat(thrown).isInstanceOf(IllegalArgumentException.class);
-            assertThat(thrown).hasMessage("Git branch of name 'refs/heads/develop' not found.");
+
+            assertThat(thrown).isInstanceOf(SkipExecutionException.class);
+            assertThat(thrown).hasMessageContaining(Property.referenceBranch.getDefaultValue());
         }
     }
 

